RecoMuon/TrackerSeedGenerator/plugins/TSGForOIDNN.cc

```diff
@@ -0,0 +1,949 @@
+    edm::FileInPath dnnMetadataPath(dnnMetadataPath_);
+    pt::read_json(dnnMetadataPath.fullPath(), metadata_);
+    tensorflow::setLogging("3");
+
+    if (useRegressor_) {
+      // use regressor
+      dnnModelPath_HB_ = metadata_.get<std::string>("HB.dnnmodel_path");
+      edm::FileInPath dnnPath_HB(dnnModelPath_HB_);
+      graphDef_HB_ = std::unique_ptr<tensorflow::GraphDef>(tensorflow::loadGraphDef(dnnPath_HB.fullPath()));
+      tf_session_HB_ = tensorflow::createSession(graphDef_HB_.get());
+
+      dnnModelPath_HLIP_ = metadata_.get<std::string>("HLIP.dnnmodel_path");
+      edm::FileInPath dnnPath_HLIP(dnnModelPath_HLIP_);
+      graphDef_HLIP_ = std::unique_ptr<tensorflow::GraphDef>(tensorflow::loadGraphDef(dnnPath_HLIP.fullPath()));
+      tf_session_HLIP_ = tensorflow::createSession(graphDef_HLIP_.get());
+
+      dnnModelPath_HLMuS_ = metadata_.get<std::string>("HLMuS.dnnmodel_path");
+      edm::FileInPath dnnPath_HLMuS(dnnModelPath_HLMuS_);
+      graphDef_HLMuS_ = std::unique_ptr<tensorflow::GraphDef>(tensorflow::loadGraphDef(dnnPath_HLMuS.fullPath()));
+      tf_session_HLMuS_ = tensorflow::createSession(graphDef_HLMuS_.get());
+    } else {
+      // use classifier (default)
+      dnnModelPath_barrel_ = metadata_.get<std::string>("barrel.dnnmodel_path");
+      edm::FileInPath dnnPath_barrel(dnnModelPath_barrel_);
+      graphDef_barrel_ = std::unique_ptr<tensorflow::GraphDef>(tensorflow::loadGraphDef(dnnPath_barrel.fullPath()));
+      tf_session_barrel_ = tensorflow::createSession(graphDef_barrel_.get());
+
+      dnnModelPath_endcap_ = metadata_.get<std::string>("endcap.dnnmodel_path");
+      edm::FileInPath dnnPath_endcap(dnnModelPath_endcap_);
+      graphDef_endcap_ = std::unique_ptr<tensorflow::GraphDef>(tensorflow::loadGraphDef(dnnPath_endcap.fullPath()));
+      tf_session_endcap_ = tensorflow::createSession(graphDef_endcap_.get());
+    }
+  }
+  produces<std::vector<TrajectorySeed> >();
+}
+
+TSGForOIDNN::~TSGForOIDNN() {
+  if (getStrategyFromDNN_) {
+    if (useRegressor_) {
+      tensorflow::closeSession(tf_session_HB_);
+      tensorflow::closeSession(tf_session_HLIP_);
+      tensorflow::closeSession(tf_session_HLMuS_);
+    } else {
+      tensorflow::closeSession(tf_session_barrel_);
+      tensorflow::closeSession(tf_session_endcap_);
+    }
+  }
+}
+
+//
+// Produce seeds
+//
+void TSGForOIDNN::produce(edm::StreamID sid, edm::Event& iEvent, edm::EventSetup const& iEventSetup) const {
+  // Initialize variables
+  unsigned int numSeedsMade = 0;
+  unsigned int layerCount = 0;
+  unsigned int hitlessSeedsMadeIP = 0;
+  unsigned int hitlessSeedsMadeMuS = 0;
+  unsigned int hitSeedsMade = 0;
+  unsigned int hitDoubletSeedsMade = 0;
+
+  // Surface used to make a TSOS at the PCA to the beamline
+  Plane::PlanePointer dummyPlane = Plane::build(Plane::PositionType(), Plane::RotationType());
+
+  // Read ESHandles
+  edm::ESHandle<MagneticField> magfieldH = iEventSetup.getHandle(t_magfieldH_);
```

Comment:
related to https://github.com/cms-sw/cmssw/pull/35237#discussion_r715262803
```suggestion
  const MagneticField& magfield = iEventSetup.getData(t_magfieldH_);
```
and then replace `magfieldH.product()` in other parts of the code with `&magfield`

Reply:
Done!